decoder/vaapidecoder_vp8.cpp

```diff
@@ -83,52 +83,14 @@ YamiStatus VaapiDecoderVP8::ensureContext()
        so, m_confiBuffer width/height may update upon key frame
        3. we don't care m_configBuffer graphicBufferWidth/graphicBufferHeight for now,
        since that is used for android
-     */
+    */
 
-    DEBUG("got frame size: %d x %d", m_frameHdr.width, m_frameHdr.height);
     m_frameWidth = m_frameHdr.width;
     m_frameHeight = m_frameHdr.height;
-
-    // only reset va context when there is a larger frame
-    if (m_configBuffer.width < m_frameHdr.width
-        || m_configBuffer.height < m_frameHdr.height) {
-        resetContext = true;
-        INFO("frame size changed, reconfig codec. orig size %d x %d, new size: %d x %d", m_configBuffer.width, m_configBuffer.height, m_frameHdr.width, m_frameHdr.height);
-        m_configBuffer.width = m_frameHdr.width;
-        m_configBuffer.height = m_frameHdr.height;
-        m_configBuffer.surfaceWidth = m_configBuffer.width;
-        m_configBuffer.surfaceHeight = m_configBuffer.height;
-        if (m_hasContext)
-            status = VaapiDecoderBase::terminateVA();
-        m_hasContext = false;
-
-        if (status != YAMI_SUCCESS)
-            return status;
-    } else if (m_videoFormatInfo.width != m_frameHdr.width
-        || m_videoFormatInfo.height != m_frameHdr.height) {
-        // notify client of resolution change, no need to reset hw context
-            INFO("frame size changed, reconfig codec. orig size %d x %d, new size: %d x %d\n", m_videoFormatInfo.width, m_videoFormatInfo.height, m_frameHdr.width, m_frameHdr.height);
-            m_videoFormatInfo.width = m_frameHdr.width;
-            m_videoFormatInfo.height = m_frameHdr.height;
-            // XXX, assume graphicBufferWidth/graphicBufferHeight are hw resolution, needn't update here
-            return YAMI_DECODE_FORMAT_CHANGE;
-    }
-
-    if (m_hasContext)
-        return YAMI_SUCCESS;
-
-    DEBUG("Start VA context");
-    status = VaapiDecoderBase::start(&m_configBuffer);
-
-    if (status != YAMI_SUCCESS)
-        return status;
-
-    m_hasContext = true;
-
-    if (resetContext)
+    if (setFormat(m_frameWidth, m_frameHeight, m_frameWidth, m_frameHeight, 8)) {
```

Comment:
1. why change this from 3 to 8.
2. could we define a enum or const static value for this?